proposals/0011-renditions-list.md

```diff
@@ -0,0 +1,172 @@
+
+```js
+// assuming a video is already loaded
+const video = document.querySelector('video');
+
+const videoTrack = video.videoTracks[0];
+const videoRenditions = videoTrack.renditions;
+
+// only keep renditions enabled that are of height less than or requal to 720
+Array.from(videoRenditions).forEach(rendition => {
+    rendition.enabled = rendition.height <= 720;
+});
+```
+
+To change the selected rendition, you can set the `selectedIndex` on the renditions list.
+```js
+// assuming a video is already loaded
+const video = document.querySelector('video');
+
+const videoTrack = video.videoTracks[0];
+const videoRenditions = videoTrack.renditions;
+
+// switch to the 3rd rendition
+videoRenditions.selectedIndex = 2;
+```
+
+Changing the `selectedIndex` manually will trigger a `change` event on the renditions list. The ABR algorithm changing the selected rendition should also trigger a `change` event on the list object.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Below is WebIDL definitions for the changes and some explanation around it.
+
+```ts
+partial interface VideoTrack {
+  readonly attribute VideoRenditionList renditions;
+};
+```
+
+`VideoRenditionList` is similar to [`VideoTrackList`](https://html.spec.whatwg.org/multipage/media.html#videotracklist).
+The main difference aside from naming is the addition of a read-write `selectedIndex` property.
+This value will default to `-1` when no rendition is selected, such as before metadata has loaded.
+Otherwise, it will reflect the index of the currently selected rendition.
+Setting the value to a number will request a rendition change to that rendition.
+
+When the rendition changes, either via manually setting the `selectedIndex`, or automatically via the ABR algorithm, a `change` event should be triggered.
+
+```ts
+interface VideoRenditionList : EventTarget {
+  readonly attribute unsigned long length;
+  getter VideoRendition (unsigned long index);
+  VideoRendition? getRenditionById(DOMString id);
+  attribute long selectedIndex;
+
+  attribute EventHandler onchange;
+  attribute EventHandler onaddrendition;
+  attribute EventHandler onremoverendition;
+};
+```
+
+Then each `VideoRendition` is includes some metadata on the rendition, like width, height, bitrate, and codec.
+It also includes a boolean to determine whether this rendition is enabled or not.
+This allowed the ABR algorithms to know whether they should be able to switch to this rendition or not.
+
+If the currently active rendition is disabled, it should request a rendition change.
+
+```ts
+interface VideoRendition {
+  readonly attribute DOMString id;
+  readonly attribute unsigned long width;
+  readonly attribute unsigned long height;
+  readonly attribute unsigned long bitrate;
+  readonly attribute unsigned long frameRate;
+  readonly attribute unsigned long codec;
+  attribute boolean enabled;
```

Comment:
this makes more sense to be in line with the track API and be named `selected`.

`enabled` is used in the AudioTrack where multiple audio tracks can be enabled.

`selected` is used when only 1 track can be selected like a VideoTrack.

when a rendition is chosen for a video track, the old rendition is deselected hence... 
the name `selected`